cmake/modules/RootConfiguration.cmake

```diff
@@ -531,6 +531,9 @@ endif()
 if(found_stdstringview)
   CHECK_CXX_SOURCE_COMPILES("#include <string_view>
      int main() { size_t pos; std::string_view str; std::stod(str,&pos); return 0;}" found_stod_stringview)
+  if(CMAKE_CUDA_STANDARD GREATER_EQUAL CMAKE_CXX_STANDARD)
+    set(cudahasstdstringview define)
```

Comment:
Doesn't that lead to a failing build with CUDA 10?

Reply:
no, why?

Reply:
Then I am confused.   

The new code states that "if the regular compiler supports std::string_view then nvcc does too" 
but in CudaMatrix.h the old code stays the opposite "even-though the regular compiler supports std::string_view, nvcc does not".

I.e. it seems that the PR essentially remove the line 24-27 of CudaMatrix.h ... which assumingly were there to support (now) older version of the CUDA toolkit.

What am I missing?

Reply:
I think it's a bit different, the new code says that only if the user specifically set the NVCC C++ standard to greater/newer than the host compiler one, only then assume the NVCC supports the same feature support specifically for `std::string_view`. While this is not exhausting all options, I think practically, if you request CMAKE_CUDA_STANDARD=17, then you are saying - yes, NVCC supports C++17 - which by definition means it should support `std::string_view`. However, I am open to suggestion on how to do it otherwise and how can it be different.

Reply:
As you can see in the cmake code right before, we do not assume that a given standard/compiler combo supports std::string_view but rather probe for it.   So technically a given compiler "could" support std::string_view with the standard set at C++11.

Reply:
oh, wait, now I see it - [this](https://github.com/root-project/root/blob/7405aa457b819aad22cece0e82166a049f1f1922/cmake/modules/SearchInstalledSoftware.cmake#L1460) is the problem. Well, then we have to think of something to test as it currently fails miserably on CUDA 11. Does anyone know the CMAKE variable to check compile with CUDA/NVCC?

Reply:
fastest fix I can think of is - instead of assuming CMAKE_CUDA_STANDARD, to require a user to set it explicitly if `cuda=ON`. Otherwise, I don't know how to do this.